src/types.rs

```diff
@@ -941,6 +941,13 @@ impl From<Vec<u8>> for OwnedParameter {
 
 impl OwnedParameter {
     pub fn as_parameter(&self) -> Parameter { Parameter(self.0.as_ref()) }
+
+    /// Construct an `OwnedParameter` by serializing the input using its
+    /// `Serial` instance.
+    pub fn new<D: Serial>(data: &D) -> Self { Self(to_bytes(data)) }
+
+    /// Construct an `OwnedParameter` from the raw, serialized, data.
+    pub fn new_raw(data: Vec<u8>) -> Self { Self(data) }
```

Comment:
There is already a `From` implementation that does exactly this so I would rather remove this.